libgalois/src/analytics/random_walks/random_walks.cpp

```diff
@@ -158,22 +99,27 @@ struct Node2VecAlgo {
           double prob = (*dist)(*generator.getLocal());
 
           //Assumption: All edges have weight 1
-          Graph::Node nbr;
-          FindSampleNeighbor(graph, n, degree, prob, &nbr);
+          Graph::Node nbr = FindSampleNeighbor(graph, n, degree, prob);
+          KATANA_LOG_ASSERT(nbr < graph.num_nodes());
 
           walk.push_back(std::move(nbr));
 
-          for (uint32_t walk_iter = 2; walk_iter <= walkLength; walk_iter++) {
+          for (uint32_t walk_iter = 2; walk_iter <= plan_.walk_length();
+               walk_iter++) {
             uint32_t curr = walk[walk_iter - 1];
             uint32_t prev = walk[walk_iter - 2];
 
+            //check if n has no neighbor
+            if (degree[curr] == 0) {
+              continue;
+            }
```

Comment:
this is not required for the symmetric case. And for un-symmetric case, if curr has degree 0, then the walk is stuck there. So probably can use `break` instead of `continue`